apps/dashboard/app/(app)/settings/root-keys/[keyId]/permissions/add-permission-for-api.tsx

```diff
@@ -40,7 +40,7 @@ export function DialogAddPermissionsForAPI(
   const [selectedApiId, setSelectedApiId] = useState<string>("");
   const selectedApi = useMemo(
     () => props.apis.find((api) => api.id === selectedApiId),
-    [selectedApiId],
+    [selectedApiId, props.apis],
```

Comment:
frequent changes to `props.apis` can hurt performance, as it's an array of objects and shallow comparison is used for dependencies. Ensure `props.apis` doesn't change often.

Reply:
@unrenamed you are right. See my comment. It was pretty weird to see these things show a lint warning. I will refactor this entirely. Archiving for now